perllib/FixMyStreet/Cobrand/Westminster.pm

```diff
@@ -88,24 +88,42 @@ sub open311_config {
     # display the road layer. Instead we'll look up the closest asset from the
     # asset service at the point we're sending the report over Open311.
     if (!$row->get_extra_field_value('USRN')) {
-        if (my $ref = $self->lookup_site_code($row)) {
+        if (my $ref = $self->lookup_site_code($row, undef, 'USRN')) {
             push @$extra, { name => 'USRN', value => $ref };
         }
     }
 
+    # Some categories require a UPRN to be set, so if the field is present
+    # but empty then look it up.
+    my $fields = $row->get_extra_fields;
+    my @uprn_field = grep { $_->{name} eq 'UPRN' } @$fields;
+    if ( scalar @uprn_field && !$row->get_extra_field_value('UPRN')) {
+        if (my $ref = $self->lookup_site_code($row, undef, 'UPRN')) {
```

Comment:
Function parameters near not the greatest. It looks like buffer is never called if you want to drop that entirely first?

Reply:
Good call, have gotten rid of the buffer param.